pkg/operator/controller/garden/garden/reconciler_reconcile.go

```diff
@@ -447,6 +400,53 @@ func (r *Reconciler) reconcile(
 			Dependencies: flow.NewTaskIDs(initializeVirtualClusterClient, waitUntilGardenerAPIServerReady, snapshotETCD),
 		})
 
+		_ = g.Add(flow.Task{
+			Name:         "Deploying fluent-operator",
+			Fn:           c.fluentOperator.Deploy,
+			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager, deployFluentCRD),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Deploying fluent-operator CustomResources",
+			Fn:           c.fluentOperatorCustomResources.Deploy,
+			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager, deployFluentCRD),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Deploying fluent-bit",
+			Fn:           c.fluentBit.Deploy,
+			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager, deployFluentCRD),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Deploying Vali",
+			Fn:           c.vali.Deploy,
+			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager),
+		})
+
+		_ = g.Add(flow.Task{
+			Name:         "Deploying prometheus-operator",
+			Fn:           c.prometheusOperator.Deploy,
+			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager, deployPrometheusCRD),
+		})
+		_ = g.Add(flow.Task{
+			Name: "Deploying Alertmanager",
+			Fn: func(ctx context.Context) error {
+				credentialsSecret, found := secretsManager.Get(v1beta1constants.SecretNameObservabilityIngress)
+				if !found {
+					return fmt.Errorf("secret %q not found", v1beta1constants.SecretNameObservabilityIngress)
+				}
+
+				c.alertManager.SetIngressAuthSecret(credentialsSecret)
+				return c.alertManager.Deploy(ctx)
+			},
+			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager, deployPrometheusCRD),
+		})
+		_ = g.Add(flow.Task{
+			Name: "Deploying Prometheus",
+			Fn: func(ctx context.Context) error {
+				return r.deployGardenPrometheus(ctx, log, secretsManager, c.prometheus, virtualClusterClient)
+			},
+			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager, deployPrometheusCRD, waitUntilGardenerAPIServerReady, initializeVirtualClusterClient),
+		})
```

Comment:
Nothing is dependent on these tasks, so not need to add them anywhere, or?

Reply:
Small addition: I'll make the BBX deployment dependent of the Garden Prometheus since it mounts its access secret and cannot start unless it got created. Otherwise, there should be no dependencies.

Reply:
Thanks for the explanation.